sdk/compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Upgrade.hs

```diff
@@ -129,14 +128,13 @@ checkPackageToDepth checkDepth pkg deps version upgradeInfo warningFlags mbUpgra
       case mbUpgradedPkg of
         Nothing -> pure ()
         Just (upgradedPkg, upgradingDeps) -> do
-            let upwnavToExternalDep UpgradedPkgWithNameAndVersion { upwnavPkgId, upwnavPkg } = ExternalPackage upwnavPkgId upwnavPkg
             depsMap <- checkUpgradeDependenciesM deps (upgradedPkg : upgradingDeps)
-            checkPackageBoth checkDepth Nothing (pkg, map upwnavToExternalDep deps) ((upwnavPkgId upgradedPkg, upwnavPkg upgradedPkg, map upwnavToExternalDep upgradingDeps), depsMap)
+            checkPackageBoth checkDepth Nothing pkg ((upwnavPkgId upgradedPkg, upwnavPkg upgradedPkg), depsMap)
 
-checkPackageBoth :: CheckDepth -> Maybe Context -> (LF.Package, [ExternalPackage]) -> ((LF.PackageId, LF.Package, [ExternalPackage]), DepsMap) -> TcPreUpgradeM ()
```

Comment:
The changes in this file were introduced in 512a521 and reverted in e25ec9e, could we add them back? We only needed to remove the warning, but these refactoring are helpful and would eliminate a potential bug further down the line.